webapp/src/components/workspaceSwitcher/workspaceSwitcher.tsx

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+import React, {useState} from 'react'
+
+import './workspaceSwitcher.scss'
+import {generatePath, useHistory, useRouteMatch} from 'react-router-dom'
+
+import {IWorkspace} from '../../blocks/workspace'
+import ChevronDown from '../../widgets/icons/chevronDown'
+
+import WorkspaceOptions from './workspaceOptions'
+
+type Props = {
+    activeWorkspace: IWorkspace
+}
+
+const WorkspaceSwitcher = (props: Props): JSX.Element => {
+    const history = useHistory()
+    const match = useRouteMatch()
+
+    const [showMenu, setShowMenu] = useState<boolean>(false)
+    const [selectClosed, setSelectClosed] = useState<boolean>(false)
+
+    return (
+        <div className={'WorkspaceSwitcherWrapper'}>
+            <div
+                className='WorkspaceSwitcher'
+                onClick={() => {
+                    if (selectClosed) {
+                        setSelectClosed(false)
+                        setShowMenu(false)
+                    } else {
+                        setShowMenu(true)
+                    }
+                }}
+            >
+                <span>{props.activeWorkspace.title}</span>
+                <ChevronDown/>
+            </div>
+            {
+                showMenu &&
+                <WorkspaceOptions
+                    activeWorkspaceId={props.activeWorkspace.id}
+                    onClose={() => {
+                        setSelectClosed(true)
+                    }}
+                    onChange={(workspaceId: string) => {
+                        const newPath = generatePath(match.path, {...match.params, workspaceId})
```

Comment:
This is not totally true. This was this way in other cases because we wanted to preserve the current path, because it can change a lot and there is not an easy way to define what is the right path, in this case, because in shared mode shouldn't be shown, and in personal server shouldn't be shown, you can assume that this url is going to be always `/workspace/:workspaceId`. In any case, if you want to keep the `generatePath` approach, you should remove the boardId and viewId from the current `match.params`, if not you are generating a route that is not correct (boardId and viewId from another workspace).

Reply:
Done.